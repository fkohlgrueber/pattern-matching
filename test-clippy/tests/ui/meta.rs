
macro_rules! pattern {
    ($($tokens:tt)*) => {
        // empty
    };
}

macro_rules! abc {
    ($($tokens:tt)*) => {
        // empty
    };
}

fn main() {
    // ...
}

pattern!{
    something: Expr = 
        Lit(_) | _
}

pattern!{
    something: Expr = 
        _ | Lit(_)
}

pattern!{
    something: Expr = 
        _ | Array(_+ Lit(_){5})
}


pattern!{
    something: Expr = 
        Lit(_)
}

abc!{
    something: Expr = 
        Lit(_) | _
}

pattern!{
    bla
}

