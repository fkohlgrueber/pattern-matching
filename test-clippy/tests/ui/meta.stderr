error: One side of the pattern has no effect because _ matches averything.
  --> $DIR/meta.rs:18:1
   |
LL | / pattern!{
LL | |     something: Expr = 
LL | |         Lit(_) | _
LL | | }
   | |_^
   |
   = note: #[forbid(pre_lint)] on by default

error: One side of the pattern has no effect because _ matches averything.
  --> $DIR/meta.rs:23:1
   |
LL | / pattern!{
LL | |     something: Expr = 
LL | |         _ | Lit(_)
LL | | }
   | |_^

error: One side of the pattern has no effect because _ matches averything.
  --> $DIR/meta.rs:28:1
   |
LL | / pattern!{
LL | |     something: Expr = 
LL | |         _ | Array(_+ Lit(_){5})
LL | | }
   | |_^

error: aborting due to 3 previous errors

